src/rgw/driver/rados/rgw_bucket.cc

```diff
@@ -2574,8 +2574,14 @@ int RGWMetadataHandlerPut_BucketInstance::put_check(const DoutPrefixProvider *dp
       }
     }
     bci.info.layout.current_index.layout.type = rule_info.index_type;
+
+    // set versioning on bucket during creation on archive zone
+    if (bihandler->driver->get_zone()->get_tier_type() == "archive") {
+      bci.info.flags = (bci.info.flags & ~BUCKET_VERSIONS_SUSPENDED) | BUCKET_VERSIONED;
+    }
```

Comment:
since this is repeated in both if/else blocks now, can we move it out so it's not repeated?

Reply:
oops